src/extension.ts

```diff
@@ -33,24 +41,49 @@ class PSCompletionItem extends CompletionItem {
 }
 
 class PSCompletionProvider implements CompletionItemProvider {
-  constructor(private snippets: [Snippet]) {}
+  constructor(private snippets: Snippets) {}
 
   provideCompletionItems(): ProviderResult<CompletionItem[]> {
     return Object.keys(snippets).map(key => new PSCompletionItem(snippets[key]))
   }
 }
 
 export async function activate(context: ExtensionContext) {
-  let registerCompletionProvider = languages.registerCompletionItemProvider(
-    'javascript',
-    new PSCompletionProvider(snippets as [Snippet]),
-    '*'
-  )
+  async function workspaceFoldersChange(folders: WorkspaceFolder[]) {
+    let config
+    if (folders) {
+      config = await prettier.resolveConfig(folders[0].uri.path)
+    } else {
+      config = workspace.getConfiguration(PRETTIER_EXTENSION)
+    }
+
+    let formattedSnippets = formatSnippets(
+      snippets,
+      { tokens: [TABSTOP, PLACEHOLDER] },
+      config
+    )
+
+    if (disposable) {
+      disposable.dispose()
+    }
+
+    disposable = languages.registerCompletionItemProvider(
+      'javascript',
+      new PSCompletionProvider(formattedSnippets),
+      '*'
+    )
+  }
+
+  let folders = workspace.workspaceFolders
+  await workspaceFoldersChange(folders)
+
+  // TODO: Somehow handle workspaces
```

Comment:
`onDidChangeActiveTextEditor` event and reference to last `rootPath` might help here.